### ISSUE TITLE
unhead is missing from server build dependencies

### ISSUE THREAD

**Issue by geminigeek** [harlan-zw/nuxt-seo#192]
### Describe the bug

unhead is missing from build dependencies, in ./output/server/package.json,  however build is not failing . but we have runtime errors, 

to reproduce 
1. clone repo 
2. run "pnpm build" we will get errors ( check below )
3. comment "@nuxtjs/seo"  in nuxt.config.ts , run pnpm build everything works !

i ran a build on my raspberry pi4 , same issue on linux

build time error
```
"unhead" is imported by ".nuxt/dist/server/server.mjs", but could not be resolved – treating it as an external dependency.
```

'pnpm preview' error 

```
[nuxt] [request error] [unhandled] [500] Cannot find package 'unhead' imported from F:\node-local\nuxt-apps\<redacted>\.output\server\chunks\app\server.mjs
```

### Reproduction

https://github.com/geminigeek/nuxt-seo-unhead-error

### System / Nuxt Info

```Shell
------------------------------
- Operating System: Windows_NT
- Node Version:     v20.10.0
- Nuxt Version:     3.10.2
- CLI Version:      3.10.1
- Nitro Version:    2.8.1
- Package Manager:  pnpm@8.15.2
- Builder:          -
- User Config:      devtools, telemetry, site, modules
- Runtime Modules:  @pinia/nuxt@0.5.1, @vueuse/nuxt@10.8.0, @nuxtjs/seo@2.0.0-rc.8
- Build Modules:    -
------------------------------
```

**Comment by alexanderivn** [comment]
@geminigeek for temporary workaround, you can disable seoExperiment in `nuxt.config.ts`


`nuxt.config.ts`
```

  seoExperiments: {
    enabled: false
  },
```

Then try to run the build again.

**Comment by harlan-zw** [comment]
I think this is a hoisting issue with nuxt, manually including the dependency is the recommended solution, similar to how you need to manually depend on a vue version

**Comment by geminigeek** [comment]
> @geminigeek for temporary workaround, you can disable seoExperiment module in `nuxt.config.ts`
> 
> ```
>   seoExperiments: {
>     enabled: false
>   },
> ```
> 
> OR
> 
> by installing unhead
> 
> `pnpm add unhead`
> 
> Then try to run the build again.

this worked, thanks!

**Comment by Dino-Kupinic** [comment]
Installing unhead worked but I still get this huge error/warning log when building

![image](https://github.com/harlan-zw/nuxt-seo/assets/87943435/501a8be5-02c1-421a-bc21-7f87634377d5)

**Comment by harlan-zw** [comment]
This should hopefully be fixed with 2.0.0-rc.9.
